tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py

```diff
@@ -0,0 +1,219 @@
+from itertools import combinations
+
+import json
+import pytest
+from unittest import mock
+import functools
+
+from mlflow.entities.model_registry import RegisteredModelTag
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_uc_registry_messages_pb2 import (
+    CreateRegisteredModelRequest,
+    UpdateRegisteredModelRequest,
+    DeleteRegisteredModelRequest,
+    GetRegisteredModelRequest,
+    SearchRegisteredModelsRequest,
+)
+from mlflow.store._unity_catalog.registry.rest_store import UcModelRegistryStore
+from mlflow.utils.proto_json_utils import message_to_json
+from mlflow.utils.rest_utils import MlflowHostCreds
+
+
+@pytest.fixture(scope="module", autouse=True)
+def request_fixture():
+    with mock.patch("requests.request") as request_mock:
+        response = mock.MagicMock()
+        response.status_code = 200
+        response.text = "{}"
+        request_mock.return_value = response
+        yield request_mock
+
+
+def mock_http_request(f):
```

Comment:
Can we import `mock_http_request_200` from `tests.helper_functions` and use it?

Reply:
Thanks for the pointer, I ended up adding a decorator version of this helper function to `tests.helper_functions` to avoid having `with` blocks in each of the tests, and using that

Reply:
I'm the one who introduced decorator approach to remove `with`, but at some point I realized decorating a test function with `mock.patch` can be confusing:

```python
@mock.patch("foo.bar")
def test_baz():
    a()
    b()
    c()

# Which functions require `foo.bar` to be patched?
```

`with mock.patch(...)` can clarify what needs mocks and can minimize the scope of mocking.

```python
def test_baz(...):
    with mock.patch(...):
        a()
    b()
    c()
```